python/EBS_SNAPSHOT_PUBLIC_RESTORABLE_CHECK/EBS_SNAPSHOT_PUBLIC_RESTORABLE_CHECK.py

```diff
@@ -0,0 +1,421 @@
+        evaluations.append(evaluation)
+    return evaluations
+
+# Function to obtain all public Amazon EBS snapshots
+def get_public_snapshots(ec2_client, owner_id):
+    snapshots = []
+    next_token = None
+    snapshots_result = {}
+    # Initiating while loop to cycle through describe_snapshots() function calls
+    while True:
+        try:
+            # If next_token is None; make first call
+            if not next_token:
+                snapshots_result = ec2_client.describe_snapshots(OwnerIds=[owner_id], RestorableByUserIds=['all'], MaxResults=1000)
+            else:
+                snapshots_result = ec2_client.describe_snapshots(OwnerIds=[owner_id], NextToken=next_token, MaxResults=1000)
+            if snapshots_result['ResponseMetadata']['HTTPStatusCode'] != 200:
+                return(False, snapshots)
+        except Exception as boto3_exception:
+            print(boto3_exception)
+            return(False, snapshots)
+        snapshots.extend(snapshots_result['Snapshots'])
+        # If NextToken is present in snapshots_result, assign NextToken for next API call
+        if 'NextToken' in snapshots_result:
+            next_token = snapshots_result['NextToken']
+        else:
+            return(True, snapshots)
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+    ec2_client = get_client("ec2", event)
+    public_snapshots_result = get_public_snapshots(ec2_client, json.loads(event['invokingEvent'])['awsAccountId'])

```

Comment:
`event['accountId']` instead of `json.loads(event['invokingEvent'])['awsAccountId']`